tests/monitoring_test.go

```diff
@@ -0,0 +1,534 @@
+				}, metricPollingTimeout, metricPollingInterval).Should(BeNumerically("==", originalCronMetricVal+i))
+			}
+			By("Ensure metric value decrements when crons are cleaned up")
+			for i := 1; i < numCrons+1; i++ {
+				err = f.CdiClient.CdiV1beta1().DataImportCrons(f.Namespace.Name).Delete(context.TODO(), fmt.Sprintf("cron-test-%d", i), metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() int {
+					return getMetricValue(f, "kubevirt_cdi_dataimportcron_outdated_total")
+				}, metricPollingTimeout, metricPollingInterval).Should(BeNumerically("==", expectedFailingCrons-i))
+			}
+		})
+	})
+
+	Context("Prometheus Rule configuration", func() {
+		It("[test_id:8259] Alerts should have all the required annotations", func() {
+			promRule := getPrometheusRule(f)
+			for _, group := range promRule.Spec.Groups {
+				if group.Name == "cdi.rules" {
+					for _, rule := range group.Rules {
+						if rule.Alert != "" {
+							Expect(rule.Annotations).ToNot(BeNil())
+							checkForRunbookURL(rule)
+							checkForSummary(rule)
+						}
+					}
+				}
+			}
+		})
+
+		It("[test_id:8812] Alerts should have all the required labels", func() {
+			promRule := getPrometheusRule(f)
+			for _, group := range promRule.Spec.Groups {
+				if group.Name == "cdi.rules" {
+					for _, rule := range group.Rules {
+						if rule.Alert != "" {
+							Expect(rule.Labels).ToNot(BeNil())
+							checkForSeverityLabel(rule)
+							checkForHealthImpactLabel(rule)
+							checkForPartOfLabel(rule)
+							checkForComponentLabel(rule)
+						}
+					}
+				}
+			}
+		})
+	})
+
+	Context("[rfe_id:7101][crit:medium][vendor:cnv-qe@redhat.com][level:component] Alert tests", func() {
+		var originalPromRule *promv1.PrometheusRule
+
+		BeforeEach(func() {
+			originalPromRule = getPrometheusRule(f)
+		})
+
+		AfterEach(func() {
+			setPrometheusRule(f, originalPromRule)
+		})
+
+		It("[test_id:7962] CDIOperatorDown alert firing when operator scaled down", func() {
+			deploymentName := "cdi-operator"
+			By("Scale down operator so alert will trigger")
+			originalReplicas := scaleDeployment(f, deploymentName, 0)
+			Eventually(func() bool {
+				dep, err := f.K8sClient.AppsV1().Deployments(f.CdiInstallNs).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return dep.Status.Replicas == 0
+			}, 20*time.Second, 1*time.Second).Should(BeTrue())
+
+			setPrometheusRuleAlertDuration(f, originalPromRule, "CDIOperatorDown", "0m")
+			waitForPrometheusAlert(f, "CDIOperatorDown")
+
+			By("Ensuring original value of replicas restored")
+			scaleDeployment(f, deploymentName, originalReplicas)
+			err := utils.WaitForDeploymentReplicasReady(f.K8sClient, f.CdiInstallNs, deploymentName)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should throw CDIDataVolumeUnusualRestartCount alert when a DataVolume object restarts more than three times", func() {
```

Comment:
So this test is particularly pricey because we are at the mercy of k8s exponential backoff (waits until 3 restarts)
That is one of the reasons I am advocating adding the alert check for existing tests
I think it's fine if we don't bring this one into monitoring_test.go and just keep it in https://github.com/kubevirt/containerized-data-importer/blob/ca32eba41c93728bb16b607c169054e8c2f60855/tests/import_test.go#L1004